src/server/frontend_wayland/xdg_shell_stable.cpp

```diff
@@ -40,7 +40,7 @@ namespace mir
 namespace frontend
 {
 
-class XdgSurfaceStable : mw::XdgSurface
+class XdgSurfaceStable : public mw::XdgSurface
```

Comment:
I don't think we ever require that `XdgSurfaceStable` is-a `mw::XdgSurface`, do we?

I know it's somewhat weird to use private inheritance, but is it a good idea to make this change? (Actual question)

Reply:
Creating a `wayland::Weak` out of a surface requires a public inheritence chain from `LifetimeTracker`.